docs/docs/themes/shadowing.md

```diff
@@ -0,0 +1,215 @@
+---
+title: Shadowing
+---
+
+
+Gatsby themes introduce a concept called "shadowing". This feature allows users to override a component, or any other file, in the `src` directory to customize it.
```

Comment:
while it feels like a detail, maybe we should include the fact that only modules that are included in the webpack bundle are shadowed?

```suggestion
Gatsby themes introduce a concept called "shadowing". This feature allows users to replace a file in the `src` directory that is included in the webpack bundle with their own implementation. This works for React components, pages in `src/pages`, JSON files, TypeScript files, as well as any other imported file (such as `.css`) in your site.
```